File: engine/model/src/main/scala/tools/packet/login/TemporaryBan.scala
package tools.packet.login

import net.opcodes.SendOpcode
import tools.packet.PacketInput

class TemporaryBan(private var _timestampUntil: Long, private var _reason: Byte) extends PacketInput {
  def timestampUntil: Long = _timestampUntil

  def reason: Byte = _reason

  override def opcode(): SendOpcode = SendOpcode.LOGIN_STATUS
}

File: engine/engine-model/src/main/java/tools/packet/login/TemporaryBan.java
package tools.packet.login;

import net.opcodes.SendOpcode;
import tools.packet.PacketInput;

public record TemporaryBan(Long timestampUntil, Byte reason) implements PacketInput {
   @Override
   public SendOpcode opcode() {
      return SendOpcode.LOGIN_STATUS;
   }
}